google/cloud/aiplatform/models.py

```diff
@@ -75,12 +86,30 @@ def _get_model(self, model_name: str) -> gca_model.Model:
         Returns:
             model: Managed Model resource.
         """
-        resource_name = ModelServiceClient.model_path(
-            self.project, self.location, model_name
+
+        # Fully qualified model name, i.e. "projects/.../locations/.../models/12345"
+        valid_name = utils.extract_fields_from_resource_name(
+            resource_name=model_name, resource_noun="models"
         )
 
-        # TODO(b/170954330) add optional instantiation if resource path given
-        model = self.api_client.get_model(name=resource_name)
+        # Partial model name (i.e. "12345") with known project and location
+        if (
+            not valid_name
+            and utils.validate_id(model_name)
+            and (self.project or initializer.global_config.project)
```

Comment:
Perhaps break up this expression since:
1. If both (self.project or initializer.global_config.project) are unset, then the user sees the error:
`ValueError("Please provide a valid model name or ID")` which is misleading.

2. If both (self.location or initializer.global_config.location) are unset, then the user sees the error:
`ValueError("Please provide a valid model name or ID")` which is misleading.

Reply:
`initializer.global_config.location` will always return the default "us-central1" if not set so we don't need to add that condition here.

Additionally, `initializer.global_config.project` will throw an exception if project is not found when being invoked and will be handled below with creating model path so that condition can also be removed here.

* Note, that there is an additional failure mode for `initializer.global_config.project` that is not currently caught: Default project_id has not been set in the environment. I will have changes to raise in that situation in the upcoming CustomTrainingJob PR.